docs/step-by-step.md

```diff
@@ -18,48 +21,135 @@ const factory = new RuntimeClientFactory({
 const chatbot = factory.createClient();
 ```
 
-2. Call `.start()` to begin a conversation session with the Voiceflow app.
-3. Store the result of `.start()` into `context`. The `context` is a `Context` object, which is a snapshot of the conversation's current state and it contains useful information like Voiceflow variables.
-4. Output the response by iterating over `context.getResponse()` and logging `trace.payload.message`. A **trace** is a piece of the entire response.
+2. Define an event handler for the `TraceType.SPEAK` event. This event fires whenever the `RuntimeClient` receives a `SpeakTrace` from the Voiceflow app. A **trace** is a piece of the entire response. 
+
+```js
+chatbot.on(TraceType.SPEAK, (trace) => {
+  console.log(`message, event handler = ${trace.payload.message}`)
+});
+```
+
+3. Call `.start()` to begin a conversation session with the Voiceflow app. 
+4. Store the result of `.start()` into `context`. The `context` is a `Context` object, which is a snapshot of the conversation's current state and it contains useful information like Voiceflow variables.
+5. Output the traces manually by iterating over `context.getTrace()` and logging `trace.payload.message`. This is alternate way of working with the Voiceflow's app's response.
 
 ```js
 (async () => {
   const context = await chatbot.start();
  
-  const traces = context.getResponse();
-  traces.forEach(trace => console.log(trace.payload.message));
+  const traces = context.getTrace();
+  traces.forEach(trace => {
+    if (trace.type === TraceType.SPEAK) {
```

Comment:
Unless this has been fixed recently, I've found that I've needed to include `trace.payload?.message` in the if statement as the payload could be undefined, which would cause this to crash.